File: ElectronicsStore.API/Models/Product.cs
using System.ComponentModel.DataAnnotations;

namespace ElectronicsStore.API.Models
{
    public class Product
    {
        public int Id { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        [Required]
        public string Description { get; set; } = string.Empty;

        [Required]
        [Range(0, double.MaxValue)]
        public decimal Price { get; set; }

        public string ImageUrl { get; set; } = string.Empty;

        [Required]
        public string Category { get; set; } = string.Empty;

        [Range(0, int.MaxValue)]
        public int Stock { get; set; }

        [Range(0, 5)]
        public double Rating { get; set; }

        public int Reviews { get; set; }
    }
}

File: src/app/model/Product.ts
import { ProductSpecifications } from './CatalogProduct';

export interface Product {
  id: string;
  name: string;
  title: string;
  description: string;
  price: number;
  stockQuantity: number;
  category: string;
  brand: string;
  specifications: ProductSpecifications;
  color: string;
  imageUrl: string;
  createdAt: Date;
  updatedAt: Date;
  rating?: number;
  reviews?: number;
} 